herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -1714,6 +1768,43 @@ public TableSpaceCheckpoint(
         }
     }
 
+    //this method returns a map with all scan values (record numbers , table digest,digestType, next autoincrement value, table name, tablespacename, query used for table scan )
+    public TableChecksum createAndWriteTableCheksum(TableSpaceManager tableSpaceManager, String tableSpace, String tableName) throws IOException, DataScannerException {
+        CommitLogResult pos;
+        boolean lockAcquired = false;
+        StatementEvaluationContext context = StatementEvaluationContext.DEFAULT_EVALUATION_CONTEXT();
+        LOGGER.log(Level.INFO, "Create and write table checksum");
+        if (context.getTableSpaceLock() == 0) {
+            long lockStamp = acquireWriteLock("checkDataConsistency_" + tableName);
+            context.setTableSpaceLock(lockStamp);
+            lockAcquired = true;
+        }
+        try {
+            AbstractTableManager tablemanager = tableSpaceManager.getTableManager(tableName);
+            if (tableSpaceManager == null) {
+                throw new TableSpaceDoesNotExistException(String.format("Tablespace %s does not exist.", tableSpace));
+            }
+            if (tablemanager == null || tablemanager.getCreatedInTransaction() > 0) {
+                throw new TableDoesNotExistException(String.format("Table %s does not exist.", tablemanager));
+            }
+            TableChecksum scanResult = TableDataChecksum.createChecksum(tableSpaceManager.getDbmanager(), null, tableSpaceManager, tableSpace, tableName);
+            ObjectMapper mapper = new ObjectMapper();
```

Comment:
Use a static final instance. ObjectMapper is stateless

Reply:
Right, I move it as a private global variable.
Can it make sense?

Reply:
yes